browser/ui/views/tabs/brave_tab_strip.cc

```diff
@@ -43,6 +43,15 @@ BraveTabStrip::BraveTabStrip(std::unique_ptr<TabStripController> controller)
 
 BraveTabStrip::~BraveTabStrip() = default;
 
+bool BraveTabStrip::IsVerticalTabsFloating() const {
+  DCHECK(ShouldShowVerticalTabs())
+      << "This method should be called when vertical tab strip is enabled";
+
+  const auto* prefs = controller_->GetProfile()->GetPrefs();
+  return prefs->GetBoolean(brave_tabs::kVerticalTabsFloatingEnabled) &&
+         prefs->GetBoolean(brave_tabs::kVerticalTabsCollapsed);
```

Comment:
These conditions make me confusing. When floating mode is enabled and it's collapsed, is vertical tab floating?

According to the below state, floating and collapsed are different state, but this method returns true when collapsed is also true. `Floating` in this method name and `kFloating` from below are different one?
```
  enum class State {
    kCollapsed,
    kFloating,
    kExpanded,
  };
```

Reply:
Yeah, Even I was confused 😭

the pref kVerticalTabsFloatingEnabled is bound to the context menu, which enable/disables the feature entirely. And kVerticalTabsCollapsed pref is for default state.

So, if kVerticalTabsFloatingEnabled is enabled, State could be transitioned from kCollapsed to kFlaoting. But otherwise, the state can't be kFloating.

And kVerticalTabsCollapsed changes, the state is either kCollapsed or kExpanded depends on the pref value.

Reply:
Ah, I see. if `kVerticalTabsCollapsed` is false, it's expanded mode.
I thought `kVerticalTabsCollapsed` is toggle between collapsed and floating mode :)